server/testdriver/core/src/main/java/org/infinispan/server/test/api/HotRodTestClientDriver.java

```diff
@@ -31,18 +33,32 @@ public HotRodTestClientDriver(TestServer testServer, TestClient testClient) {
          // Configure admin user by default
          builder.security().authentication().username(TestUser.ADMIN.getUser()).password(TestUser.ADMIN.getPassword());
       }
-
       this.clientConfiguration = builder;
    }
 
    /**
-    * Provide a custom client configuration to connect to the server
+    * Provide a custom client configuration to connect to the server. This method always overrides the configured marshaller
+    * so if a marshaller other than the default {@link ProtoStreamMarshaller} is required, a subsequent call to
+    * {@link #withMarshaller(Class)} is required.
     *
     * @param clientConfiguration
     * @return the current {@link HotRodTestClientDriver} instance with the client configuration override
     */
    public HotRodTestClientDriver withClientConfiguration(ConfigurationBuilder clientConfiguration) {
       this.clientConfiguration = clientConfiguration;
+      // Explicitly configure the ProtoStreamMarshaller as the jboss-marshalling module is on the classpath
+      this.clientConfiguration.marshaller(ProtoStreamMarshaller.class);
+      return this;
+   }
+
+   /**
+    * The {@link Marshaller} to be used by the client.
+    *
+    * @param marshallerClass
+    * @return the current {@link HotRodTestClientDriver} instance with the Marshaller configuration override
+    */
+   public HotRodTestClientDriver withMarshaller(Class<? extends Marshaller> marshallerClass) {
```

Comment:
@tristantarrant I couldn't think of a better way of accomplishing this whilst keeping `infinispan-jboss-marshalling` on the classpath and making it so by default the tests utilise ProtoStream. In 12 we should probably remove the automatic configuration of `GenericJbossMarshaller` when `infinispan-jboss-marshalling` is present on the classpath.

Reply:
https://issues.redhat.com/browse/ISPN-11974